src/lib.rs

```diff
@@ -28,6 +30,25 @@ pub enum DecodeError<'a> {
     },
 }
 
+impl<'a> fmt::Display for DecodeError<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            DecodeError::Invalid { valid_prefix, invalid_sequence, remaining_input, } => {
+                write!(f, "invalid byte sequence (preceding valid string of length {}: '{}'; \
```

Comment:
I parse this as "invalid sequence that precedes a valid string, which is backward. Could you rephrase this? Also, maybe reduce repetition. Something like: `found invalid byte sequence {:02x?} after {} valid bytes, followed by {} more unprocessed bytes`

(Similarly below for incomplete)

Reply:
Done. I also hit it with `cargo fmt` again (which is the reason for the syntactic churn).